caddy/examples/config_example.json

```diff
@@ -0,0 +1,172 @@
+{
+    "admin": {
+        "disabled": true
+    },
+    "logging": {
+      "logs": {
+        "default": {"level":"DEBUG", "encoder": {"format":"console"}}
+      }
+    },
+    "apps": {
+        "http": {
+            "servers": {
+                "metrics": {
+                    "listen": [
+                        ":9091"
+                    ],
+                    "routes": [
+                        {
+                            "match": [
+                                {
+                                    "path": [
+                                        "/metrics"
+                                    ]
+                                }
+                            ],
+                            "handle": [
+                                {
+                                    "disable_openmetrics": true,
+                                    "handler": "metrics"
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        },
+        "layer4": {
+            "servers": {
+                "1": {
+                    "listen": [
```

Comment:
This is not actually what I meant, and it won't work if someone uses prefixes that triggers the HTTP DPI. Shadowsocks and the webserver should not share a port.

The websocket reverse proxy should be in the HTTP handler in the **http app**, not layer4. It can contain a handler that is a layer4 handler, but in the websocket handler.

Something like:

```json
    "apps": {
        "http": {
            "servers": {
                "metrics": {
                    "listen": [":443"],
                    "routes": [
                        {
                            "match": [{"path": ["/tcp"]}],
                            "handle": [
                                {
                                    "handler": "ws2stream"
                                    "transport": {
                                        "handler": "shadowsocks",
                                        "keys": [
                                            ...
                                        ]
                                    }
                                }
                            ]
                        },
                        {
                            "match": [{"path": ["/udp"]}],
                            "handle": [
                                {
                                    "handler": "ws2packet"
                                    "transport": {
                                        "handler": "shadowsocks",
                                        "keys": [
                                            ...
                                        ]
                                    }
                                }
                            ]
                        }
                    ]
                }
            }
        },
```

But we need to figure out how to share the layer4 handler with the websocket tcp and udp code and the layer4 code.

Perhaps we need to define named layer4 handlers that we can then just refer by name.